client/src/components/layouts/global.css

```diff
@@ -187,6 +187,10 @@ button.close:focus {
   background-color: transparent;
 }
 
+:focus {
+  outline: var(--focus-outline-color) solid 2px;
+}
```

Comment:
You don't need to create `--focus-outline-color` if the color is the same in both themes.



---

This selector isn't applied, looking at the devtools it doesn't reach the interactive elements, because of bootstrap higher specificity class.

The outline doesn't have enough offset like mentioned in the issue.

if you fix the specificity issue from bootstrap, can you remove the elements that have the same outline styles to clear some of the bloating css.

Reply:
I don't remember seeing an offset mentioned in the issue? Can you point it out? I remember the width of the outline being specified though.

Reply:
True, I have read the issue again and it doesn't include offset, you can ignore offset part.

Reply:
Additionally in hindsight , should I be using using border-color over outline in the css?

Reply:
border-color has width and height that are respected, which can break the layout and shift items in the page. outline on the other had doesn't have respected width and height.

Reply:
Pardon me Sboony, but are you sure my focus selector isn't being applied? 

It appears that focus selectors don't actually show up in the inspector.

![Example](https://i.imgur.com/1qDk9Rg.png)

There's also a JS fiddle I want to show you locally as a reference to what I'm talking about.

Reply:
can you be more specific of what is the question?

Reply:
I was trying to point that `:focus` selectors don't show up in the inspector and was trying to point out the possibility that maybe they're being applied but we can't easily find it out other than by hitting tab and seeing them the outline show up by hitting tab. 

My question was how are we supposed to be one hundred percent that my selector wasn't being applied. Like is there a way to find out using dev tools or not?

Reply:
In the css section of the devtools there are `:hov` button next to the search click it and click `:focus` or `:focus-visible` if you have changed it already.

Whatever interactive element you select will have focus indicator, your change isn't applied because of its specificity.


Yes you are technically applying the outline, but what is the point if the applied change isn't used, because `class:focus` elements which have the issue have a higher specificity, so your change is ignored when it's needed.

Reply:
Thank you for teaching me something new.

Reply:
Okay where is it not being applied where it should? That will help narrow down my search.

Reply:
In https://www.freecodecamp.org/learn/2022/responsive-web-design/learn-typography-by-building-a-nutrition-label/step-66, you can find a couple of areas that still have different outline colors cased by bootstrap.

> Also notice that we have several different focus outline colors in the header/breadcrumbs. While not technically a WCAG violation I think having a standard color for all of them would be better (not just aesthetically but also for accessibility).

![image](https://user-images.githubusercontent.com/88248797/223337911-162e0562-8f29-4329-9c20-9039a544b65f.png)

> pretty much non-existent on others. 

These are in the setting page, any toggle button doesn't have any focus indicator.

Reply:
To save time, because we want to apply this everywhere with the same color and stuff and maybe delete the classes in bootstrap, it will break the outline in specific location, but it's better to add specific fixes, than try to overwrite everything, and overwrite it again whenever something new is added.